src/server/game/Scripting/ScriptReloadMgr.cpp

```diff
@@ -67,6 +67,9 @@ namespace fs = boost::filesystem;
 #ifdef _WIN32
     #include <windows.h>
     #define HOTSWAP_PLATFORM_REQUIRES_CACHING
+#elif __APPLE__
+    #include <dlfcn.h>
+    #define HOTSWAP_PLATFORM_REQUIRES_CACHING
```

Comment:
I was playing around with '.gm' command from GM script, and making changes in notification system (eg. GM mode on/off) so I could have visual feedback on functionality of loaded script.
And without caching script used by the core is not reloaded at all, even though whole compile/reload procedure successfully completes after script file change.

Also , OS X definitely does not lock loaded shared libraries.